### PR TITLE
Move Worker API to App Level and deprecating old ServiceWorker API

### PR DESCRIPTION
**Pull request by gokulavasan**
Bamboo : http://builds.cask.co/browse/CDAP-DUT758

JIRA : https://issues.cask.co/browse/CDAP-1450

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/service/AbstractService.java [hunk 1]
```diff
@@ -74,7 +73,7 @@ protected void addHandlers(Iterable<? extends HttpServiceHandler> handlers) {
    * Add a worker to the Service.
    * @param worker for the service.
    */
-  @Beta
+  @Deprecated
```
Comment (ParthGandhi) [inline cdap-api/src/main/java/co/cask/cdap/api/service/AbstractService.java]: Add a link to the new API.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/worker/DefaultWorkerConfigurer.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  private String name;
+  private String description;
+  private Resources resource;
+  private int instances;
+  private Map<String, String> properties;
+  private Set<String> datasets;
+
+  public DefaultWorkerConfigurer(Worker worker) {
+    this.name = worker.getClass().getSimpleName();
```
Comment (ParthGandhi) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/worker/DefaultWorkerConfigurer.java]: Probably also want to have a `setName()` method to allow setting the name inside `configure`. Currently we allow that by having `addWorker(String name, Worker worker)` inside the Service, but also good to have that here.

**Comment by ParthGandhi** [comment]
For each API marked `@Deprecated` add a link to the replacement API in the javadoc.

**Comment by ParthGandhi** [comment]
Just some doc fixes. Otherwise LGTM.

**Comment by gokulavasan** [comment]
@ParthGandhi Addressed your comments. Please take a look.

**Comment by ParthGandhi** [comment]
:+1:

**Pull request merged**